src/MLJModels.jl

```diff
@@ -72,6 +72,7 @@ include("metadata.jl")
 
 # read in the metadata:
 metadata_file = joinpath(srcdir, "registry", "Metadata.toml")
+Base.include_dependency(metadata_file)
```

Comment:
Not sure what the point of this may be. We've never needed it in the past. I wonder if this is responsible for the fail on julia 1.11

Reply:
@Yuan-Ru-Lin Can you please try removing this line?

Reply:
The line was for `MLJModels.jl` to be precompiled after I changed `Metadata.toml`. I recommend we keep it there because anyone who wants to add new models will benefit from it.

As for the CI error, it seems that Julia 1.11.2 expects `Statistics@1.11.2` but the latest version is `Statistics@1.11.1`.

```
...
│ To reproduce this CI run locally run the following from the same repository state on julia version 1.11.2:
│
│ `import Pkg; Pkg.test(;coverage=true, julia_args=["--check-bounds=yes", "--compiled-modules=yes"], force_latest_compatible_version=false, allow_reresolve=true)`
...
ERROR: LoadError: Unsatisfiable requirements detected for package Statistics [10745b16]:
 Statistics [10745b16] log:
 ├─possible versions are: 1.11.0 - 1.11.1 or uninstalled
 ├─restricted to versions [0.0.0, 1] by project [e5070b9b], leaving only versions: 1.11.0 - 1.11.1
 │ └─project [e5070b9b] log:
 │   ├─possible versions are: 0.0.0 or uninstalled
 │   └─project [e5070b9b] is fixed to version 0.0.0
 └─restricted to versions 1.11.2 by an explicit requirement — no versions left
```

Reply:
> The line was for MLJModels.jl to be precompiled after I changed Metadata.toml. 

Ah, I see. You're right that's a useful function doing exactly what we want- force complilation if the metadata changes. 

@DilumAluthge Do you think the fail referenced above is a Julia bug? I couldn't find any reported matching it.

Reply:
@ablaom I think I've identified the problem. I'll work on a PR to fix it.

Reply:
@ablaom Fixed by #576.

Reply:
Thanks @DilumAluthge. See my comment at #576.